### PR TITLE
Sched note wdog

### PR DESCRIPTION
**Pull request by XuNeo**
## Summary

1. Add note support for wdog module.
2. Update systemview correspondingly, using timer enter/exit.

## Impact

No.

## Testing

Tested with stm32f429i-disco board.

### Configure

`tools/configure.sh stm32f429i-disco:systemview`

`echo 123 > /dev/ttyR0` will trigger some wdog usage, so you can see the note is recorded.

sysview-wdog

### CODE REVIEW

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



The PR summary is OK, but lacks details on the exact changes made. The impact assessment is incomplete.  Here's a breakdown of what needs improvement:

**Summary - Needs Improvement**

* **More Detail:** Instead of just saying "Add note support," specify what kind of notes are added, how they are triggered, and what information they provide.  For example: "Add systemview notes to track watchdog feeding, timeout events, and configuration changes."
* **Code Location:**  Mention the specific files or modules modified (e.g., "Changes made to the wdog driver and the SystemView integration layer").

**Impact -  Incomplete**

* **New Feature:**  Explicitly state that this adds a new feature (systemview watchdog visualization).
* **User Impact:** While the core functionality doesn't change, users utilizing SystemView for debugging will now have additional insights into the watchdog.  This should be mentioned.
* **Build Impact:**  Does enabling this feature require any specific build options or configurations?
* **Documentation:** Does the PR include documentation updates to explain the new notes and how to use them effectively?

**Testing - Insufficient**

* **Host Details:** Provide the operating system and compiler version used for building.
* **Target Specificity:** Mention the NuttX configuration used on your stm32f429i-disco (e.g., `stm32f429i-disco:nsh`).
* **Logs Missing:** The provided example shows a screenshot, but the actual testing logs (both before and after) are absent.  Include relevant log snippets demonstrating the changes in SystemView output.

**In summary,** the PR needs more detail in the summary and impact sections, along with complete testing information to meet the NuttX requirements.

File: include/nuttx/sched_note.h [hunk 1]
```diff
@@ -386,6 +390,13 @@ struct note_irqhandler_s
   uint8_t nih_irq;              /* IRQ number */
 };
 
+struct note_wdog_s
+{
+  struct note_common_s nmm_cmn;      /* Common note parameters */
```
Comment (anchao) [inline include/nuttx/sched_note.h:395]: What does meaning of **cmm** ?

Comment (XuNeo) [reply include/nuttx/sched_note.h:395]: `nmm` means `note common memory` according to this commit.

https://github.com/apache/nuttx/pull/13847/commits/530783008fe66f40c2e2c1bc328df014eca8864a

Comment (XuNeo) [reply include/nuttx/sched_note.h:395]: May rename to `nwd`?  And correspondingly `nhp` for heap?

**Review by anchao — APPROVED** [review]
LGTM

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**